### PR TITLE
queued operations and ts

### PR DESCRIPTION
**Pull request by tbillington**
Queues async operations instead of blocking on them.

I've never seen `ncc` before but it said it supports TS out of the box 🤷 

I haven't ran this because I don't know how to run actions locally, but theoretically it will function :D pls test

### CODE REVIEW

File: src/post.ts [hunk 1]
```diff
@@ -21,14 +21,14 @@ const SKIPPED_FILES = [
  * @param {string} manifestPath;
  * @returns {string}
  */
-async function locateTarget(manifestPath) {
+async function locateTarget(manifestPath: string): Promise<string> {
```
Comment (BD103) [inline src/post.ts:24]: Do `async` functions need to return `Promise<T>`, or can they just return `T`?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "compilerOptions": {
+    "target": "es2015",
+    "moduleResolution": "node",
+    "strict": true,
+  }
+}
```
Comment (BD103) [inline tsconfig.json:7]: (Nit) Could you add a newline to the end of this file? I try to be consistent across the entire project. :)

File: src/post.ts [hunk 3]
```diff
@@ -66,32 +66,38 @@ async function main() {
...
+                    }
+
+                    await fs.rm(filePath);
+                } else {
+                    core.debug(`Skipped ${filePath} because it was accessed after timestamp.`);
+                }
+            }));
     }
+
+    await Promise.all(operationHandles);
```
Comment (BD103) [inline src/post.ts:100]: Cool! I never realized you could do this! (Then again, my Javascript knowledge is sparse.)

File: src/post.ts [hunk 4]
```diff
@@ -66,32 +66,38 @@ async function main() {
     const targetPath = await locateTarget(manifestPath);
     core.info(`Sweeping files from ${targetPath}.`);
 
+    const operationHandles: Promise<void>[] = [];
```
Comment (BD103) [inline src/post.ts:69]:
```suggestion
    // An array of promises that will be awaited on all at once.
    const operationHandles: Promise<void>[] = [];
```

File: src/post.ts (hunk 1 above)
Comment (tbillington) [reply src/post.ts:24]: All async functions always return promises, just how they work :)

**Comment by BD103** [comment]
Thank you!

**Pull request merged**